pkg/vsphere/session/session.go

```diff
@@ -193,45 +195,55 @@ func (s *Session) Populate(ctx context.Context) (*Session, error) {
 
 	finder := s.Finder
 
+	log.Debug("vSphere resource cache populating...")
 	s.Datacenter, err = finder.DatacenterOrDefault(ctx, s.DatacenterPath)
 	if err != nil {
-		errs = append(errs, err.Error())
+		errs = append(errs, fmt.Sprintf("Failure finding dc (%s): %s", s.DatacenterPath, err.Error()))
 	} else {
 		finder.SetDatacenter(s.Datacenter)
+		log.Debugf("Cached dc: %s", s.DatacenterPath)
 	}
 
+	finder.SetDatacenter(s.Datacenter)
+
 	s.Cluster, err = finder.ComputeResourceOrDefault(ctx, s.ClusterPath)
 	if err != nil {
-		errs = append(errs, err.Error())
+		errs = append(errs, fmt.Sprintf("Failure finding cluster (%s): %s", s.ClusterPath, err.Error()))
 	}
+	log.Debugf("Cached cluster: %s", s.ClusterPath)
 
 	s.Datastore, err = finder.DatastoreOrDefault(ctx, s.DatastorePath)
 	if err != nil {
-		errs = append(errs, err.Error())
+		errs = append(errs, fmt.Sprintf("Failure finding ds (%s): %s", s.DatastorePath, err.Error()))
 	}
+	log.Debugf("Cached ds: %s", s.DatastorePath)
 
 	s.Host, err = finder.HostSystemOrDefault(ctx, s.HostPath)
 	if err != nil {
 		if _, ok := err.(*find.DefaultMultipleFoundError); !ok || !s.IsVC() {
-			errs = append(errs, err.Error())
+			errs = append(errs, fmt.Sprintf("Failure finding host (%s): %s", s.HostPath, err.Error()))
 		}
 	}
+	log.Debugf("Cached host: %s", s.HostPath)
 
 	if s.NetworkPath != "" {
 		s.Network, err = finder.NetworkOrDefault(ctx, s.NetworkPath)
 		if err != nil {
-			errs = append(errs, err.Error())
+			errs = append(errs, fmt.Sprintf("Failure finding net (%s): %s", s.NetworkPath, err.Error()))
 		}
 	}
+	log.Debugf("Cached net: %s", s.NetworkPath)
 
 	s.Pool, err = finder.ResourcePoolOrDefault(ctx, s.PoolPath)
 	if err != nil {
-		errs = append(errs, err.Error())
+		errs = append(errs, fmt.Sprintf("Failure finding pool (%s): %s", s.PoolPath, err.Error()))
 	}
+	log.Debugf("Cached pool: %s", s.PoolPath)
 
 	if len(errs) > 0 {
+		log.Debugf("Multiple errors popluating vSphere cache: (%d)", len(errs))
```

Comment:
typo "popluating".  While you're in there, s/Multiple errors/Errors/ in the message (there may only be 1 error)